rust/azure_iot_operations_services/src/azure_device_registry/client.rs

```diff
@@ -0,0 +1,1007 @@
+                CreateDetectedAssetCommandInvoker::new(
+                    application_context.clone(),
+                    client.clone(),
+                    &aep_name_command_options,
+                ),
+            ),
+            notify_on_asset_update_command_invoker: Arc::new(
+                NotifyOnAssetUpdateCommandInvoker::new(
+                    application_context.clone(),
+                    client.clone(),
+                    &aep_name_command_options,
+                ),
+            ),
+            create_asset_endpoint_profile_command_invoker: Arc::new(
+                CreateDiscoveredAssetEndpointProfileCommandInvoker::new(
+                    application_context,
+                    client.clone(),
+                    &AepCommandOptionsBuilder::default()
+                        .topic_token_map(HashMap::from([(
+                            "ex:connectorClientId".to_string(),
+                            client.client_id().to_string(),
+                        )]))
+                        .build()
+                        .expect("Statically generated options should not fail."),
+                ),
+            ),
+            aep_update_event_telemetry_dispatcher,
+            asset_update_event_telemetry_dispatcher,
+        }
+    }
+
+    /// Retrieves an asset endpoint profile from a Azure Device Registry service.
+    ///
+    /// # Arguments
+    /// * `aep_name` - The name of the asset endpoint profile.
+    /// * `timeout` - The duration until the Client stops waiting for a response to the request, it is rounded up to the nearest second.
+    ///
+    /// Returns a [`AssetEndpointProfile`] if the the endpoint was found.
+    ///
+    /// # Errors
+    /// [`struct@Error`] of kind [`InvalidArgument`](ErrorKind::InvalidArgument)
+    /// if the `timeout` is zero or > `u32::max`, or there is an error building the request.
+    ///
+    /// [`struct@Error`] of kind [`ServiceError`](ErrorKind::ServiceError)
+    /// if there is an error returned by the ADR Service.
+    ///
+    /// [`struct@Error`] of kind [`AIOProtocolError`](ErrorKind::AIOProtocolError)
+    /// if there are any underlying errors from the AIO RPC protocol.
+    pub async fn get_asset_endpoint_profile(
+        &self,
+        aep_name: String,
+        timeout: Duration,
+    ) -> Result<AssetEndpointProfile, Error> {
+        let command_request = rpc_command::invoker::RequestBuilder::default()
+            .topic_tokens(HashMap::from([("ex:aepName".to_string(), aep_name)]))
+            .timeout(timeout)
+            .build()
+            .map_err(|e| Error(ErrorKind::InvalidArgument(e.to_string())))?;
+
+        let result = self
+            .get_asset_endpoint_profile_command_invoker
+            .invoke(command_request)
+            .await;
+
+        match result {
+            Ok(response) => Ok(response.payload.asset_endpoint_profile.into()),
+            Err(e) => Err(Error(ErrorKind::AIOProtocolError(e))),
```

Comment:
This can be simplified - first to remove the assignment of the result
```suggestion
        match self
            .get_asset_endpoint_profile_command_invoker
            .invoke(command_request)
            .await {
            Ok(response) => Ok(response.payload.asset_endpoint_profile.into()),
            Err(e) => Err(Error(ErrorKind::AIOProtocolError(e))),
```
but also you might be able to add into/from implementations for AIOProtocolError into your ErrorKind::AIOProtocolError so that you can simplify it to this (although maybe you'd want to add back in the result struct at this point to maintain readability)
```suggestion
        Ok(self
            .get_asset_endpoint_profile_command_invoker
            .invoke(command_request)
            .await?.payload.asset_endpoint_profile.into())
```